x-pack/plugins/ml/public/application/components/import_export_jobs/export_jobs_flyout/export_job_warning_callout.tsx

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+
+import { EuiCallOut, EuiText, EuiAccordion, EuiSpacer } from '@elastic/eui';
+import type { JobDependencies } from './jobs_export_service';
+
+interface Props {
+  jobs: JobDependencies;
+}
+
+export const ExportJobDependenciesWarningCallout: FC<Props> = ({ jobs: allJobs }) => {
+  const usingCalendars = allJobs.some((j) => j.calendarIds.length > 0);
+  const usingFilters = allJobs.some((j) => j.filterIds.length > 0);
+
+  const jobs = allJobs.filter(
```

Comment:
Small nit: Might be worth creating `jobsUsingFilters` and `jobsUsingCalendars` lists when initially going through the jobs list here and passing those to the appropriate sub components to avoid re-filtering in `CalendarJobList` and `FilterJobList`.

Reply:
Good point, this initial filter isn't actually needed. so i'll remove it and just pass the full list of jobs about.

Reply:
actually, this is more fiddly than i first thought.
I need that first filter as its supplying a filtered list to the `getTitle` function.
I can use a reducer to produce three filtered lists
```typescript
  const [jobs, jobsWithCalendars, jobsWithFilters] = allJobs.reduce(
    (acc, job) => {
      const usingCalendars = job.calendarIds.length > 0;
      const usingFilters = job.filterIds.length > 0;
      if (usingCalendars || usingFilters) {
        acc[0].push(job);
        if (usingCalendars) {
          acc[1].push(job);
        }
        if (usingFilters) {
          acc[2].push(job);
        }
      }
      return acc;
    },
    [[], [], []] as JobDependency[][]
  );
```
although that's quite fancy looking, it's not as simple as just using an extra filter.

Reply:
Went with the fancy looking change https://github.com/elastic/kibana/pull/107416/commits/6b533f4b5adf3c972a291bd2c32333c7b2ebb2ea